src/lib/mongoBackend/mongoQueryContext.cpp

```diff
@@ -195,10 +203,13 @@ void addContextProviders(ContextElementResponseVector& cerV, ContextRegistration
 
     if (cr.contextRegistrationAttributeVector.size() == 0)
     {
-      /* Registration without attributes */
-      for (unsigned int jx = 0; jx < cr.entityIdVector.size(); ++jx)
+      if (!limitReached)
       {
-        addContextProviderEntity(cerV, cr.entityIdVector.get(jx), cr.providingApplication);
+        /* Registration without attributes */
+        for (unsigned int jx = 0; jx < cr.entityIdVector.size(); ++jx)
+        {
+          addContextProviderEntity(cerV, cr.entityIdVector.get(jx), cr.providingApplication);
```

Comment:
cr.entityIdVector[jx] ?

If the vector doesn't have the operator [, then lets create it.
'jx' is not a very good variable name. eIx is better.